### ISSUE TITLE
🐛 [firebase_ui_oauth_google] "An unknown error occurred" with google sign in 

### ISSUE THREAD

**Issue by X-Wei** [firebase/flutterfire#10166]
## Bug report

**Describe the bug**

### Steps to reproduce

pubspec deps:

```yaml
environment:
  sdk: '>=2.18.6 <3.0.0'

dependencies:
  cloud_firestore: ^4.2.0
  firebase_auth: ^4.2.1
  firebase_core: ^2.4.0
  firebase_crashlytics: ^3.0.7
  firebase_performance: ^0.9.0+7
  firebase_ui_auth: ^1.1.3
  firebase_ui_firestore: ^1.1.1
  firebase_ui_oauth_google: ^1.0.10
```

I added google provider in the firebase_ui_auth with

```dart
final _providers = <AuthProvider<AuthListener, AuthCredential>>[
  GoogleProvider(clientId: GOOGLE_CLIENT_ID),
  EmailAuthProvider(),
];
```

The web app works fine, but with **android**, I get ""An unknown error occurred" red message when clicking on google sign in btn...

I searched and found https://github.com/firebase/flutterfire/issues/8384 which was closed in october.

### Flutter doctor

Run `flutter doctor` and paste the output below:

<details><summary>Click To Expand</summary>

```sh
$ flutter upgrade
Flutter is already up to date on channel stable
Flutter 3.3.10 • channel stable • https://github.com/flutter/flutter.git
Framework • revision 135454af32 (7 days ago) • 2022-12-15 07:36:55 -0800
Engine • revision 3316dd8728
Tools • Dart 2.18.6 • DevTools 2.15.0

$ flutter doctor
Doctor summary (to see all details, run flutter doctor -v):
[✓] Flutter (Channel stable, 3.3.10, on Ubuntu 20.04.5 LTS 5.4.0-135-generic, locale en_US.UTF-8)
[✓] Android toolchain - develop for Android devices (Android SDK version 33.0.1)
[✓] Chrome - develop for the web
[✓] Linux toolchain - develop for Linux desktop
[✓] Android Studio (version 2021.3)
[✓] VS Code (version 1.74.1)
[✓] Connected device (2 available)
[✓] HTTP Host Availability

• No issues found!
```

</details>

---

**Comment by darshankawar** [comment]
@X-Wei 
Can you provide the complete error log that is shown ?
It seems the package has just been published very recently. Did you follow all the instructions mentioned in its readMe and also check its implementation ?

**Comment by X-Wei** [comment]
oh sorry for the false alert, I've figured it out.

In app1, I forgot to add the SHA1 fingerprints to the android app on firebase console, causing this failure.

In app2, I registered the same package id in 2 firebase projects, and caused confusion, I followed the instructions [here](https://support.google.com/firebase/answer/6401008?hl=en) and deleted the other app's OAuth client ID, now it's working fine.
